docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,358 @@
+assume there's a `Block` structure, let's call it `v1`. If in the future there's a need to modify this structure that is not compatible with the
+previous data structure, then the database would support a `v2` schema for `ResultBlock` and an `index` table could determine the criteria on which
+data structure should be used for inserting or querying data.
+
+There's also a possibility that the data structure could be stored in parallel with the normalized data but in one blob, e.g. in a `jsonb` field
+that contains all the information needed to return the client, this could be an optimized way to serve the data but could add to the storage
+requirements. This could improve the performance for reading data from the database. An alternative approach to this problem that can offer
+similar results would be to use a "caching" layer.
+
+
+### [RPC server instance](#rpc-instance)
+
+The **RPC server instance** is a node that runs the RPC API process for the data companion. This server instance provides an RPC API (`/v1`) with
+the same JSONRPC methods of the full node JSONRPC endpoint. The RPC Companion service will expose the same JSONRPC methods and will accept the same request types and
+return wire compatible responses (should match the same response as the equivalent full node JSONRPC endpoint).
+
+The **RPC server instance**, when serving a particular request, retrieves the required data from the database in order to
+fulfill the request. The data should be serialized in a way that makes it wire compatible with the CometBFT JSONRPC endpoint.
+
+The RPC service endpoint should be exposed through an external load-balancer service such as Cloudflare or AWS ELB, or
+a server running its own load balancer mechanism (e.g. nginx).
+
+The RPC clients should make requests to the **RPC Companion** server instances through this load balancer.
+
+The **RPC server instance** will also implement logic for the proxied requests to the full node. It should properly handle the proxy requests and responses from the full node.
+
+The **RPC Companion** should support the `https` protocol in order to support a secure endpoint access. It's recommended that
+the `https` support is provided by the load balancer but in case there's a single **RPC Companion** server instance, having an
+option to use `https` is important and it's backwards compatible with the existing CometBFT RPC that supports that.
+
+
+## Consequences
+
+### Positive
+
+- Alternative and optional **RPC Companion** that is more scalable and reliable with a higher query throughput.
+- Less backpressure on the full node that is running consensus.
+- Possibility for future additional (e.g a `/v2`) with additional methods not available in the `/v1` endpoint.
+- Can act as a basis for users to create better and faster indexers and analytics solutions.
+- Possibility to turn off indexing on the full node if that can be offload to the RPC Companion.
```

Comment:
RPC Companion or Data Companion?

Reply:
make it more clear that is an external data storage with indexing